source/server/server.cc

```diff
@@ -535,8 +535,16 @@ void InstanceImpl::initialize(const Options& options,
   // Once we have runtime we can initialize the SSL context manager.
   ssl_context_manager_ = createContextManager("ssl_context_manager", time_source_);
 
-  const bool use_tcp_for_dns_lookups = bootstrap_.use_tcp_for_dns_lookups();
-  dns_resolver_ = dispatcher_->createDnsResolver({}, use_tcp_for_dns_lookups);
+  auto dns_resolver_options =
+      envoy::config::core::v3::DnsResolverOptions(bootstrap_.dns_resolver_options());
+  // Field bool `use_tcp_for_dns_lookups` will be deprecated in future. To keep supporting earlier
+  // implementation of control plane APIs only accept this value if `use_tcp_for_dns_lookups`
+  // is not set via dns_resolver_options but is set `true` in the field bool
+  // `use_tcp_for_dns_lookups`.
+  if (bootstrap_.use_tcp_for_dns_lookups() && !dns_resolver_options.has_use_tcp_for_dns_lookups()) {
+    dns_resolver_options.mutable_use_tcp_for_dns_lookups()->set_value(true);
+  }
```

Comment:
check please.

Reply:
Same explanation as previous. I want to use _**bootstrap_.use_tcp_for_dns_lookups**_ if and only if _dns_resolver_options_ doesn't have any value set for _use_tcp_for_dns_lookups_.